tock/hours/views.py

```diff
@@ -943,6 +944,23 @@ def get_queryset(self, queryset=None):
         return sorted_fiscal_years
 
 
+    def get_context_data(self, **kwargs):
+        # Call the base implementation first to get a context
+        context = super().get_context_data(**kwargs)
+        # Get today's date to figure out the fiscal year it is in
+        today = localdate()
+        # The year will be used to determine which year reporting to end with
+        year = ReportingPeriod.get_fiscal_year_from_date(today)
+        context['fiscal_years'] = []
+        for year in range(2017, year + 1):
+            year_dates = dict()
```

Comment:
I probably would have written all of this to the dict in one shot in order to (hopefully) be a bit more readable...
```
context['fiscal_years'].append({
     'year': year,
     'start_date': blahblahblah,
    'end_date': BobLoblaw
})
```
But it's fine the way you have it here too

Reply:
This is actually a really confusing piece of code now that you pointed out.  I also was using `year` as each instance in the for loop as well as a variable for the end of range.  I am rewriting it now.